vendor/github.com/rancher/norman/controller/generic_controller.go

```diff
@@ -124,26 +124,37 @@ func (g *genericController) Enqueue(namespace, name string) {
 
 func (g *genericController) AddHandler(ctx context.Context, name string, handler HandlerFunc) {
 	g.Lock()
+	defer g.Unlock()
+
+	g.generation++
 	h := &handlerDef{
 		name:       name,
 		generation: g.generation,
 		handler:    handler,
 	}
-	g.handlers = append(g.handlers, h)
-	g.Unlock()
 
-	go func() {
+	go func(gen int) {
```

Comment:
What is the purpose of generation?

Reply:
It's part of the ID and stops from duplicating the same handler

Reply:
To clarify, it's for deleting the handler after the context closes